core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
@lindong28 I see your point. The difference you mentioned occurs only if all the partitions in the request are "unknown partitions". Otherwise, even if a single partition is authorized and valid and returns a group level error, with the current logic that error is what's reported for all partitions in the request (due to [this line](https://github.com/vahidhashemian/kafka/blob/9898d665d1ab201405d66c70e3ea9710d9dcecd7/core/src/main/scala/kafka/server/KafkaApis.scala#L959)). I'll modify the PR. Thanks for your feedback again.